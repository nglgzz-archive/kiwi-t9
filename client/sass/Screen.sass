@import variables


@keyframes blink
  0%
    opacity: 1

  50%
    opacity: 1

  51%
    opacity: 0

  100%
    opacity: 0


.screen
  +flex-center
  background-color: $tom-thumb
  border-radius: 10px
  box-shadow: inset $top-left-shadow, inset $top-right-shadow
  color: $hunter-green
  font-family: 'Nokia'
  height: 290px
  justify-content: space-between
  margin: 10px
  max-width: 260px
  padding: 10px

  .actions
    display: flex
    justify-content: space-between
    width: 100%

    button
      background: none
      border: 0
      color: $hunter-green
      cursor: pointer
      font-family: inherit
      font-size: inherit
      outline: 0

  .statusbar
    align-self: flex-end


.textbox
  border: 4px solid $hunter-green
  height: 100%
  padding: 5px
  width: calc(100% - 10px)

  &-lastword::after
    animation: blink 2s infinite
    background-color: $hunter-green
    content: '-'
    display: inline
