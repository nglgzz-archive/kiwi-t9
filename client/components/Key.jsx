import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import * as Suggestions from 'actions/Suggestions';
import * as Symbols from 'actions/Symbols';
import 'sass/Key.sass';


@connect(store => ({
  digits: store.text.digits,
}))
export default class Key extends Component {
  constructor(props) {
    super(props);

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    switch (this.props.label) {
      case '1':
        this.props.dispatch(Symbols.insert());
        break;

      case '*':
        // Cycle through suggestions
        this.props.dispatch(Suggestions.next());
        break;

      case '0':
        // Finish the current word and start a new one
        this.props.dispatch(Suggestions.end());
        break;

      case '#':
        // add new word
        break;

      default:
        this.props.dispatch(Suggestions.fetch((
          this.props.digits + this.props.label
        )));
    }
  }

  render() {
    const { label, description, position } = this.props;
    const positionName = ['left', 'center', 'right'][position % 3];

    return (
      <button
        className={`key key--${positionName}`}
        onClick={this.handleClick}
      >
        <span className="key-label">{label}</span>
        <span className="key-description">{description}</span>
      </button>
    );
  }
}


Key.propTypes = {
  label: PropTypes.string.isRequired,
  position: PropTypes.number.isRequired,
  description: PropTypes.string.isRequired,
};
