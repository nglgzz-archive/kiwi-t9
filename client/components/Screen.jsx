import React, { Component } from 'react';
import { connect } from 'react-redux';
import 'sass/Screen.sass';


@connect(store => ({
  text: store.text.text,
  lastWord: store.text.lastWord,
}))
export default class Screen extends Component {
  render() {
    const { text, lastWord } = this.props;

    return (
      <div className="screen">
        <div className="statusbar">
          {(new Date()).toLocaleTimeString().slice(0, 5)}
        </div>

        <div className="textbox">
          <span>{text}</span>
          <span className="last-word">{lastWord}</span>
          <span className="cursor" />
        </div>

        <div className="actions">
          <a href="/">Send</a>
        </div>
      </div>
    );
  }
}
