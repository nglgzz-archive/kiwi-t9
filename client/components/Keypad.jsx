import React from 'react';
import Key from 'components/Key';
import keypadLayout from 'utils/keypadLayout.json';
import 'sass/Keypad.sass';


export default () => {
  const keys = keypadLayout.map(({ label, description }, index) => (
    <Key
      position={index}
      key={label}
      label={label}
      description={description}
    />
  ));

  return (
    <div className="keypad">{keys}</div>
  );
};
