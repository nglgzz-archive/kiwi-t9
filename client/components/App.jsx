import React from 'react';
import Keypad from 'components/Keypad';
import Screen from 'components/Screen';
import 'sass/App.sass';


export default () => (
  <div className="phone">
    <div className="speaker">
      <span />
      <span />
      <span />
    </div>

    <Screen />
    <Keypad />
  </div>
);
