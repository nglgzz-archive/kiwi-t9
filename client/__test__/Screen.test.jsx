import React from 'react';
import { Screen } from 'components/Screen';


describe('Screen component', () => {
  const setup = (propOverrides) => {
    const props = {
      words: 'hello world',
      ...propOverrides,
    };

    return shallow(<Screen {...props} />);
  };

  it('renders without crashing', () => {
    const wrapper = setup();
    expect(wrapper.length).toBe(1);
  });

  it('matches snapshot', () => {
    // Use fixed time for the snapshot (the component contains a clock that
    // otherwise would make the test fail every time).
    const fixedTime = new Date('10-10-2010');
    const RealDate = Date;
    // eslint-disable-next-line no-global-assign
    Date = function constructor() { return fixedTime; };

    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();

    // Restore original value of Date.
    // eslint-disable-next-line no-global-assign
    Date = RealDate;
  });

  it('dispatches TEXT_RESET', () => {
    const mockDispatch = jest.fn();
    const wrapper = setup({ dispatch: mockDispatch });

    wrapper.find('.actions--reset').simulate('click');
    expect(mockDispatch.mock.calls.length).toBe(1);
    expect(mockDispatch).toBeCalledWith({
      type: 'TEXT_RESET',
    });
  });

  it('dispatches CHAR_DELETE', () => {
    const mockDispatch = jest.fn();
    const wrapper = setup({ dispatch: mockDispatch });

    wrapper.find('.actions--delete').simulate('click');
    expect(mockDispatch.mock.calls.length).toBe(1);
    expect(mockDispatch).toBeCalledWith({
      type: 'CHAR_DELETE',
    });
  });

  it('redirect to twitter', () => {
    // https://github.com/facebook/jest/issues/890#issuecomment-295939071
    window.location.assign = jest.fn();
    const wrapper = setup();
    const url = 'https://twitter.com/home?status=hello%20world';

    wrapper.find('.actions--send').simulate('click');
    expect(window.location.assign).toBeCalledWith(expect.stringContaining(url));
  });
});
