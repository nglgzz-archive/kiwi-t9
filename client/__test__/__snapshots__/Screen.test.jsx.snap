// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen component matches snapshot 1`] = `
<div
  className="screen"
>
  <div
    className="statusbar"
  >
    <span
      className="charcounter"
    >
      129
    </span>
    <span
      className="clock"
    >
      0:00
    </span>
  </div>
  <div
    className="textbox"
  >
    hello world
  </div>
  <div
    className="actions"
  >
    <button
      className="actions--reset"
      onClick={[Function]}
    >
      Reset
    </button>
    <button
      className="actions--send"
      onClick={[Function]}
    >
      Send
    </button>
    <button
      className="actions--delete"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
